acme/authority.go

```diff
@@ -263,21 +265,33 @@ func (a *Authority) ValidateChallenge(p provisioner.Interface, accID, chID strin
 	if accID != ch.getAccountID() {
 		return nil, UnauthorizedErr(errors.New("account does not own challenge"))
 	}
+	retry := ch.getRetry()
+	if retry.Active {
+		return ch.toACME(a.db, a.dir, p)
+	}
+	retry.Mux.Lock()
+	defer retry.Mux.Unlock()
+
 	client := http.Client{
 		Timeout: time.Duration(30 * time.Second),
 	}
-	dialer := &net.Dialer{
-		Timeout: 30 * time.Second,
-	}
-	ch, err = ch.validate(a.db, jwk, validateOptions{
-		httpGet:   client.Get,
-		lookupTxt: net.LookupTXT,
-		tlsDial: func(network, addr string, config *tls.Config) (*tls.Conn, error) {
-			return tls.DialWithDialer(dialer, network, addr, config)
-		},
-	})
-	if err != nil {
-		return nil, Wrap(err, "error attempting challenge validation")
+
+	for i:=0; i < 10; i++ {
+		ch, err = ch.validate(a.db, jwk, validateOptions{
+			httpGet:   client.Get,
+			lookupTxt: net.LookupTXT,
+		})
+		if err != nil {
+			return nil, Wrap(err, "error attempting challenge validation")
+		}
+		if ch.getStatus() == StatusValid {
+			break
+		}
+		if ch.getStatus() == StatusInvalid {
+			return ch.toACME(a.db, a.dir, p)
+		}
+		duration := time.Duration(ch.getRetry().Backoffs + math.Mod(rand.Float64(), 5))
```

Comment:
Starting at 1, doubling 10 time means we end up with ~17min gap (or it might actually be ~8.5min with the logic as written since we'll only double 9 times). That's probably okay but I'd actually consider maybe scaling this into milliseconds and starting with a 300ms or 500ms gap or something, then doubling to ~1 and so on.

Reply:
This doubling I assumed would come on the next call to ValidateChallenge(), not on every loop iteration. The above comment seems to be similarly themed. Will update with backoffs increasing on each call in that case